src/equations/acoustic_perturbation_2d.jl

```diff
@@ -104,51 +148,207 @@ function source_terms_convergence_test(u, x, t, equations::AcousticPerturbationE
   omega = 2 * pi * f
 
   si, co = sincos(omega * (x[1] + x[2] - a * t))
-  tmp = v_mean[1] + v_mean[2] - a
+  tmp = v1_mean + v2_mean - a
 
   du1 = du2 = A * omega * co * (2 * c + tmp + 2/rho_mean * A * si)
   du3 = A * omega * co * (2 * c_mean^2 * rho_mean + 2 * c * tmp + 2 * A * tmp * si)
 
-  return SVector(du1, du2, du3)
+  du4 = du5 = du6 = du7 = 0.0
+
+  return SVector(du1, du2, du3, du4, du5, du6, du7)
+end
+
+
+"""
+    initial_condition_gauss(x, t, equations::AcousticPerturbationEquations2D)
+
+A Gaussian pulse in a constant mean flow. Uses the global mean values from `equations`.
+"""
+function initial_condition_gauss(x, t, equations::AcousticPerturbationEquations2D)
+  v1_prime = 0.0
+  v2_prime = 0.0
+  p_prime = exp(-4*(x[1]^2 + x[2]^2))
+
+  return SVector(v1_prime, v2_prime, p_prime, global_mean_vars(equations)...)
+end
+
+
+"""
+    initial_condition_gauss_wall(x, t, equations::AcousticPerturbationEquations2D)
+
+A Gaussian pulse, used in the `gauss_wall` example elixir in combination with
+[`boundary_condition_wall`](@ref). Uses the global mean values from `equations`.
+"""
+function initial_condition_gauss_wall(x, t, equations::AcousticPerturbationEquations2D)
+  v1_prime = 0.0
+  v2_prime = 0.0
+  p_prime = exp(-log(2) * (x[1]^2 + (x[2] - 25)^2) / 25)
+
+  return SVector(v1_prime, v2_prime, p_prime, global_mean_vars(equations)...)
+end
+
+"""
+    boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                            equations::AcousticPerturbationEquations2D)
+
+Boundary conditions for a solid wall.
+"""
+function boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                                 equations::AcousticPerturbationEquations2D)
+  # Boundary state is equal to the inner state except for the perturbed velocity. For boundaries
+  # in the -x/+x direction, we multiply the perturbed velocity in the x direction by -1.
+  # Similarly, for boundaries in the -y/+y direction, we multiply the perturbed velocity in the
+  # y direction by -1
```

Comment:
I am not sure that these are the correct boundary conditions for a wall for aeroacoustics. Where did you get this formulation from? The results look OK, though.

Reply:
I've used them before in a lecture. They were essentially given to us as "here, use these boundary conditions for this exercise", so I don't know the precise theoretical considerations behind them or if they're appropriate for aeroacoustics. The literature you gave me didn't include a description of the wall boundary conditions used there, so I decided to just try these first and since the results seemed ok to me I didn't look into it again.